core/Enclave/key_operation.cpp

```diff
@@ -967,31 +1025,37 @@ sgx_status_t ehsm_sm2_decrypt(const ehsm_keyblob_t *cmk,
  * digest mode and padding mode is optional
  * running in enclave
  * @param cmk_blob cipher block for storing keys
- * @param data data to be signed
+ * @param message data to be signed
  * @param signature used to receive signature
  * @return sgx_status_t
  */
 sgx_status_t ehsm_rsa_sign(const ehsm_keyblob_t *cmk,
-                           const ehsm_data_t *data,
+                           uint32_t algorithm,
+                           uint32_t message_type,
+                           const ehsm_data_t *message,
                            ehsm_data_t *signature)
 {
     sgx_status_t ret = SGX_ERROR_UNEXPECTED;
 
-    // verify padding mode
-    if (cmk->metadata.padding_mode != EH_PAD_RSA_PKCS1 && cmk->metadata.padding_mode != EH_PAD_RSA_PKCS1_PSS)
-        return SGX_ERROR_INVALID_PARAMETER;
-
     uint8_t *rsa_keypair = NULL;
     BIO *bio = NULL;
     RSA *rsa_prikey = NULL;
 
-    // Get Digest Mode
-    const EVP_MD *digestMode = GetDigestMode(cmk->metadata.digest_mode);
-    if (digestMode == NULL)
+    // Get padding mode and digest mode
+    const EVP_MD *digest_mode = getDigestModeFromAlgorithm((ehsm_signature_algorithm_t)algorithm);
+    if (digest_mode == NULL)
     {
-        log_d("ecall rsa_sign digest Mode error.\n");
-        return SGX_ERROR_INVALID_PARAMETER;
+        log_e("Fail to parse digest mode");
+        return ret;
+    }
+
+    int padding_mode = getPaddingModeFromSignatureAlgorithm((ehsm_signature_algorithm_t)algorithm);
+    if (padding_mode == RSA_NO_PADDING)
```

Comment:
RSA_NO_PADDING is defined as error ?

Reply:
no padding is not support, I change log to log_e("No padding is not support.");